src/main/java/frc/robot/subsystems/drive/ReefscapeVisionUtil.java

```diff
@@ -0,0 +1,98 @@
+        private static final Pose2d RED_REEF_I = TAG_11.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d RED_REEF_J = TAG_11.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+        private static final Pose2d RED_REEF_K = TAG_6.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d RED_REEF_L = TAG_6.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+
+        private static final Pose2d BLUE_REEF_A = TAG_18.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d BLUE_REEF_B = TAG_18.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+        private static final Pose2d BLUE_REEF_C = TAG_19.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d BLUE_REEF_D = TAG_19.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+        private static final Pose2d BLUE_REEF_E = TAG_20.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d BLUE_REEF_F = TAG_20.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+        private static final Pose2d BLUE_REEF_G = TAG_21.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d BLUE_REEF_H = TAG_21.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+        private static final Pose2d BLUE_REEF_I = TAG_22.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d BLUE_REEF_J = TAG_22.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+        private static final Pose2d BLUE_REEF_K = TAG_17.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_LEFT));
+        private static final Pose2d BLUE_REEF_L = TAG_17.transformBy(REEF_TO_ROBOT.plus(SCORING_POSE_OFFSET_RIGHT));
+
+        private static final List<Pose2d> LEFT_POSES_RED = Arrays.asList(
+                        RED_REEF_A, RED_REEF_C, RED_REEF_E, RED_REEF_G, RED_REEF_I, RED_REEF_K);
+
+        private static final List<Pose2d> RIGHT_POSES_RED = Arrays.asList(
+                        RED_REEF_B, RED_REEF_D, RED_REEF_F, RED_REEF_H, RED_REEF_J, RED_REEF_L);
+
+        private static final List<Pose2d> LEFT_POSES_BLUE = Arrays.asList(
+                        BLUE_REEF_A, BLUE_REEF_C, BLUE_REEF_E, BLUE_REEF_G, BLUE_REEF_I, BLUE_REEF_K);
+
+        private static final List<Pose2d> RIGHT_POSES_BLUE = Arrays.asList(
+                        BLUE_REEF_B, BLUE_REEF_D, BLUE_REEF_F, BLUE_REEF_H, BLUE_REEF_J, BLUE_REEF_L);
+
+        public static Supplier<Pose2d> getGoalPose(RobotPosition robotPosition, Supplier<Pose2d> robotPose,
+                        BooleanSupplier isBlueAlliance) {
+                Pose2d goalPose;
+                if (robotPose.get() == null) {
+                        Pose2d nullReturn = new Pose2d();
+                        return () -> nullReturn;
```

Comment:
nit - returning `Pose2d{0,0}` from here means that callers need to know that `{0,0}` means "don't go there"

this is irrelevant since the robotPose.get() can never be null (the WPILib pose estimators never return null)